app/api/i18n.v2/database/MongoTranslationsSyncDataSource.ts

```
@@ -0,0 +1,44 @@
+import { MongoDataSource } from 'api/common.v2/database/MongoDataSource';
+import { SyncDBDataSource } from 'api/odm';
+import { DeleteResult, ObjectId, OptionalId } from 'mongodb';
+import { TranslationDBO } from '../schemas/TranslationDBO';
+
+export class MongoTranslationsSyncDataSource
+  extends MongoDataSource<OptionalId<TranslationDBO>>
+  implements SyncDBDataSource<TranslationDBO>
+{
+  protected collectionName = 'translationsV2';
+
+  async save(translation: TranslationDBO): Promise<TranslationDBO> {
+    await this.getCollection().updateOne(
+      { _id: translation._id },
+      { $set: translation },
+      { upsert: true }
+    );
+    return translation;
+  }
+
+  async saveMultiple(translations: TranslationDBO[]): Promise<TranslationDBO[]> {
+    const stream = this.createBulkStream();
+    await translations.reduce(async (previous, translation) => {
+      await previous;
+      await stream.updateOne({ _id: translation._id }, { $set: translation }, true);
+    }, Promise.resolve());
+
+    await stream.flush();
+    return translations;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async get(_query: any, _select?: any, _options?: any): Promise<TranslationDBO[]> {
```

Is this unused?

---

Throw an error instead of returnin an empty array.